### PR TITLE
When invocationcounter is out of sync, do request for new ip-address

### PR DESCRIPTION
**Pull request by kroesctrl**
When invocationcounter is out of sync, do request for new ip-address

### CODE REVIEW

**Comment by kroesctrl** [comment]
https://jenkins.fdp.osgp.cloud/view/SMHE-cucumber/job/gxf-smhe-cucumber-build/145/

**Review by abolder — APPROVED** [review]

File: osgp/protocol-adapter-dlms/osgp-protocol-adapter-dlms/src/main/java/org/opensmartgridplatform/adapter/protocol/dlms/domain/factories/InvocationCounterManager.java [hunk 1]
```diff
@@ -59,21 +59,32 @@ public InvocationCounterManager(
    */
   public void initializeInvocationCounter(
       final MessageMetadata messageMetadata, final DlmsDevice device) throws OsgpException {
+    /*
+     * When the invocation counter is out of sync, the device closes the session.
```
Comment (stefanermens) [inline osgp/protocol-adapter-dlms/osgp-protocol-adapter-dlms/src/main/java/org/opensmartgridplatform/adapter/protocol/dlms/domain/factories/InvocationCounterManager.java]: Is this true for all devices? Is "some devices close the session" more correct?

File: osgp/protocol-adapter-dlms/osgp-protocol-adapter-dlms/src/main/java/org/opensmartgridplatform/adapter/protocol/dlms/domain/factories/InvocationCounterManager.java [hunk 2]
```diff
@@ -59,21 +59,32 @@ public InvocationCounterManager(
...
+     * This is done by the DlmsConnectionFactory in the method:
+     * this.domainHelperService.setIpAddressFromMessageMetadataOrSessionProvider
+     */
+    device.setIpAddress(null);
 
     this.initializeWithInvocationCounterStoredOnDevice(messageMetadata, device, null);
+
+    /*
+     * When the invocation counter has been initialized, the device closes the session.
+     * By setting the ip-address to null, the application will be forced to get a new ip-address.
```
Comment (stefanermens) [inline osgp/protocol-adapter-dlms/osgp-protocol-adapter-dlms/src/main/java/org/opensmartgridplatform/adapter/protocol/dlms/domain/factories/InvocationCounterManager.java]: This comment is the same as the previous comment. To make it clear that it's really the intention to do this twice, I suggest to change it to: "by setting the ip-address to null again".

Comment (kroesctrl) [reply osgp/protocol-adapter-dlms/osgp-protocol-adapter-dlms/src/main/java/org/opensmartgridplatform/adapter/protocol/dlms/domain/factories/InvocationCounterManager.java]: Changed

File: osgp/protocol-adapter-dlms/osgp-protocol-adapter-dlms/src/main/java/org/opensmartgridplatform/adapter/protocol/dlms/domain/factories/InvocationCounterManager.java (hunk 1 above)
Comment (kroesctrl) [reply osgp/protocol-adapter-dlms/osgp-protocol-adapter-dlms/src/main/java/org/opensmartgridplatform/adapter/protocol/dlms/domain/factories/InvocationCounterManager.java]: changed

**Review by stefanermens — APPROVED** [review]

**Pull request merged**